kernel/kernel_drivers.go

```diff
@@ -195,6 +217,20 @@ func setupModsFromComp(kernelTree, kversion string, kmodsConts []snap.ContainerP
 	}
 	logger.Noticef("depmod output:\n%s\n", string(osutil.CombineStdOutErr(stdout, stderr)))
 
+	// Change symlinks to target ones when needed
+	for _, mci := range comps {
+		if mci.Mounts.CurrentEqualsTarget() {
+			continue
+		}
+		lname := filepath.Join(compsRoot, mci.Name)
+		to := mci.Mounts.UnderTargetPath("modules", kversion)
+		// remove old link
+		os.Remove(lname)
```

Comment:
do we to check the error?
```suggestion
		if err := os.Remove(lname); err != nil && !errors.Is(err, fs.ErrNotExist) { 
			return fmt.Errorf("cannot remove old link: %w", err)
		}	
```

Reply:
the osSymlink itself will fail if the remove is relevant and fails